uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1030.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1030 {
```

Comment:
General question, we try to make all our upgraders idempotent, I think if you run this upgrader again it will repeatedly delete all option values for this group and recreate them. Do you think it would be possible to make sure it only deletes those outside of the desired options and doesn't recreate them if they exist?

Reply:
Good idea. I will do the check.

Reply:
I see that you added a check so the option value will not be created if it exists. This is a positive step but I think you should consider what happens if you run the upgrader again

```
// this will be all the option values (the new ones if run twice)
$oldSicknessReasons = civicrm_api3('OptionValue', 'get', [
  'return' => ['name'],
  'option_group_id' => 'hrleaveandabsences_sickness_reason',
]);
// these will then be disabled or deleted
```

I don't think it's worth spending a lot of time on this, because in reality upgraders will only be run once. But we have had bugs where they were run more than once, and I think it's a good strategy to make them idempotent so please keep this in mind when working on them in the future